### PR TITLE
added support for ldap users that have spaces in their names

### PR DESCRIPTION
**Pull request by jnfrmarks**
Support for ldap users. We need to do some special handling for ldap users with spaces in the name.

### CODE REVIEW

File: owncloud/owncloud.py [hunk 1]
```diff
@@ -726,20 +726,39 @@ def user_exists(self, user_name):
...
+# LDAP users can have a space in their names; ownCloud does not support
+# spaces in usernames so the space gets converted to an underscore.
+# When we search for the ldap user with a space in the name, we need
+# to remove the underscore to find the user.  To do this, if the username
+# being searched for has an underscore in it, we will search for users with
+# names that begin with the input username up to the underscore and then search
+# for the actual username in the result.  
+# If the name of the user we're looking for is ldap_user, we will search
+# for all users that have ldap in the name and then look explicitly for 
+# ldap_user in the results.
```
Comment (PVince81) [inline owncloud/owncloud.py]: Shouldn't we rather fix this in the server instead ?
If not in the users backend, then at least do this in the provisioning API module ?

CC @blizzz

**Comment by jnfrmarks** [comment]
@PVince81 

Moved the ldap logic out of owncloud.py and put it in smasbox.  Ready for review again.

**Comment by PVince81** [comment]
Can we close this PR then ?

Can you link to the matching PR in smashbox ?

**Comment by PVince81** [comment]
Ah right, you just removed the workaround, so we should keep this PR

File: owncloud/owncloud.py [hunk 2]
```diff
@@ -717,6 +717,34 @@ def delete_user(self, user_name):
 
         raise ResponseError(res)
 
+    def get_user_list(self, user_name):
+        """Gets a lit of users via provisioning API.
```
Comment (PVince81) [inline owncloud/owncloud.py]: Typo: "lit" => "list"

File: owncloud/owncloud.py [hunk 3]
```diff
@@ -717,6 +717,34 @@ def delete_user(self, user_name):
...
+        :param user_name:  substring of username to be checked 
+        :returns: a list of users matching the user_name sub string
+        :raises: ResponseError in case an HTTP error status was returned
+
+        """
+
+        res = self.__make_ocs_request(
+            'GET',
+            self.OCS_SERVICE_CLOUD,
+            'users?search=' + user_name
```
Comment (PVince81) [inline owncloud/owncloud.py]: Please encode/escape the user_name argument

Comment (jnfrmarks) [inline owncloud/owncloud.py]: Are you suggesting I pass in the user_name to the api requrst as:

user_name.encode ('utf_8')

Shouldn't the api handle that?

We should probably make the change made throughout the file for all strings being passed into the api (group names, file names, user names, etc) in which case do we want a new branch for this work?

Comment (PVince81) [inline owncloud/owncloud.py]: No, you should url encode it, see https://github.com/owncloud/pyocclient/blob/master/owncloud/owncloud.py#L650

URL encode means that spaces get converted to "%20", slashes to "%2F", etc.
This is not done automatically by the "requests" lib if you simply concatenate strings like you do here.

**Comment by PVince81** [comment]
closing as abandoned, conflicts, ...

**Pull request closed without merging**